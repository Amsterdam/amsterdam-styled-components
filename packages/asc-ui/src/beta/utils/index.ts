import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
