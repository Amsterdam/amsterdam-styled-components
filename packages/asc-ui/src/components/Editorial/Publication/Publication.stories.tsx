import React from 'react'
import { action } from '@storybook/addon-actions'
import {
  Container,
  Column,
  CustomHTMLBlock,
  Row,
  EditorialHeader,
} from '../../../index'
import DocumentCover from '../../DocumentCover'
import publicationJSON from './publication_feed.json'
import { EditorialMetaList, EditorialContent } from '../index'
import Paragraph from '../../Paragraph'

export default {
  title: 'Dataportaal/Composed/Editorial/Publication',

  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
    ),
  ],
}

export const ImplementationForDataportal = () => {
  const {
    title,
    body,
    field_publication_intro: intro,
    field_file_size: fileSize,
    field_file_type: fileType,
    filecover_url: coverUrl,
    file_url: downloadUrl,
  } = publicationJSON

  return (
    <Container>
      <Row>
        <Column
          wrap
          span={{ small: 1, medium: 2, big: 6, large: 12, xLarge: 12 }}
        >
          <Column span={{ small: 1, medium: 2, big: 6, large: 12, xLarge: 12 }}>
            <EditorialContent>
              <EditorialHeader title={title} />
              <EditorialMetaList
                fields={[
                  { id: 1, label: fileSize },
                  { id: 2, label: fileType },
                ]}
              />
            </EditorialContent>
          </Column>
          <Column span={{ small: 1, medium: 2, big: 3, large: 6, xLarge: 6 }}>
            <DocumentCover
              imageSrc={`${coverUrl}`}
              description={`Download PDF (${fileSize})`}
              onClick={action(`downloading: ${downloadUrl}`)}
            />
          </Column>
          <Column span={{ small: 1, medium: 2, big: 3, large: 6, xLarge: 6 }}>
            <EditorialContent>
              {intro && <Paragraph strong>{intro}</Paragraph>}
              <CustomHTMLBlock body={body} />
            </EditorialContent>
          </Column>
        </Column>
      </Row>
    </Container>
  )
}
