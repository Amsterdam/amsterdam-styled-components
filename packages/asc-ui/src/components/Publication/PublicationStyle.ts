import styled from '@datapunt/asc-core'
import { BlogPostStyle } from '../Blog/BlogPost'

export type Props = {}

export default styled(BlogPostStyle)<Props>``
