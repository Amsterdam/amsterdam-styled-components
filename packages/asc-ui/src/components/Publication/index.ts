import PublicationContentStyle from './PublicationContent/PublicationContentStyle'

export { default } from './Publication'
export { default as PublicationContent } from './PublicationContent'

export const PublicationStyles = {
  PublicationContentStyle,
}
