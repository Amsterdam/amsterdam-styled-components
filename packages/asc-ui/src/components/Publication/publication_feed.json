{
  "jsonapi": {
    "version": "1.0",
    "meta": {
      "links": {
        "self": {
          "href": "http:\/\/jsonapi.org\/format\/1.0\/"
        }
      }
    }
  },
  "data": [
    {
      "type": "node--publication",
      "id": "dd354ea9-5eca-4810-9c68-c5a38a90f1bb",
      "attributes": {
        "drupal_internal__nid": 41,
        "drupal_internal__vid": 49,
        "langcode": "en",
        "revision_timestamp": "2019-07-10T14:13:03+00:00",
        "revision_log": null,
        "status": true,
        "title": "Amsterdammers over het Wallengebied",
        "created": "2019-07-10T14:13:03+00:00",
        "changed": "2019-07-10T14:18:42+00:00",
        "promote": false,
        "sticky": false,
        "default_langcode": true,
        "revision_translation_affected": true,
        "path": {
          "alias": null,
          "pid": null,
          "langcode": "en"
        },
        "body": {
          "value": "\u003Cp\u003EDit rapport beschrijft de uitkomsten van de enque\u0302te over het Wallengebied onder de bewoners van Amsterdam. Bij alle onderwerpen is gekeken of er verschillen zijn tussen groepen bewoners op basis van de volgende achtergrondkenmerken:\u003C\/p\u003E\r\n\r\n\u003Cul\u003E\r\n\t\u003Cli\u003EGeslacht\u003C\/li\u003E\r\n\t\u003Cli\u003ELeeftijd\u0026nbsp;\u003C\/li\u003E\r\n\t\u003Cli\u003EMigratie-achtergrond (geen, westers of niet-westers).\u003C\/li\u003E\r\n\u003C\/ul\u003E\r\n",
          "format": "basic_html",
          "processed": "\u003Cp\u003EDit rapport beschrijft de uitkomsten van de enque\u0302te over het Wallengebied onder de bewoners van Amsterdam. Bij alle onderwerpen is gekeken of er verschillen zijn tussen groepen bewoners op basis van de volgende achtergrondkenmerken:\u003C\/p\u003E\n\n\u003Cul\u003E\u003Cli\u003EGeslacht\u003C\/li\u003E\n\t\u003Cli\u003ELeeftijd\u00a0\u003C\/li\u003E\n\t\u003Cli\u003EMigratie-achtergrond (geen, westers of niet-westers).\u003C\/li\u003E\n\u003C\/ul\u003E",
          "summary": ""
        },
        "field_file_size": "966 KB",
        "field_file_type": "pdf",
        "field_pub": null,
        "field_publication_intro": "OIS, AT5 en Het Parool hebben in overleg een korte vragenlijst opgesteld over de Wallen. De vragenlijst is afgenomen via het online-panel van OIS, telefonische interviews met een aselecte groep Amsterdammers en straatinterviews met jongeren en niet-westerse Amsterdammers, twee groepen die over het algemeen ondervertegenwoordigd zijn in enque\u0302teonderzoek. Aan het onderzoek hebben in totaal 688\u00a0Amsterdammers meegewerkt.",
        "field_publication_month": "1",
        "field_publication_source": "Onderzoek, Informatie en Statistiek",
        "field_publication_year": "2019"
      },
      "relationships": {
        "node_type": {
          "data": {
            "type": "node_type--node_type",
            "id": "e3dedc05-1de6-4e25-b6dc-37197d73ad96"
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/relationships\/node_type?resourceVersion=id%3A49"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/node_type?resourceVersion=id%3A49"
            }
          }
        },
        "revision_uid": {
          "data": {
            "type": "user--user",
            "id": "44418b09-7bfb-43ef-a488-7a03211bf407"
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/relationships\/revision_uid?resourceVersion=id%3A49"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/revision_uid?resourceVersion=id%3A49"
            }
          }
        },
        "uid": {
          "data": {
            "type": "user--user",
            "id": "44418b09-7bfb-43ef-a488-7a03211bf407"
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/relationships\/uid?resourceVersion=id%3A49"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/uid?resourceVersion=id%3A49"
            }
          }
        },
        "field_cover_image": {
          "data": {
            "type": "file--file",
            "id": "b657da23-243a-4c96-ae14-59971da69a6d",
            "meta": {
              "alt": "",
              "title": "",
              "width": 620,
              "height": 878
            }
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/relationships\/field_cover_image?resourceVersion=id%3A49"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/field_cover_image?resourceVersion=id%3A49"
            }
          }
        },
        "field_publication_file": {
          "data": {
            "type": "file--file",
            "id": "fc9b6db2-0c29-4337-b4a2-fed055646856",
            "meta": {
              "display": true,
              "description": ""
            }
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/relationships\/field_publication_file?resourceVersion=id%3A49"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb\/field_publication_file?resourceVersion=id%3A49"
            }
          }
        }
      },
      "links": {
        "self": {
          "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication\/dd354ea9-5eca-4810-9c68-c5a38a90f1bb?resourceVersion=id%3A49"
        }
      }
    }
  ],
  "included": [
    {
      "type": "file--file",
      "id": "b657da23-243a-4c96-ae14-59971da69a6d",
      "attributes": {
        "drupal_internal__fid": 79,
        "langcode": "en",
        "filename": "2019-rapportage-wallengebied.png",
        "uri": {
          "value": "public:\/\/images\/2019-rapportage-wallengebied.png",
          "url": "\/sites\/default\/files\/images\/2019-rapportage-wallengebied.png"
        },
        "filemime": "image\/png",
        "filesize": 721451,
        "status": true,
        "created": "2019-07-10T14:13:24+00:00",
        "changed": "2019-07-10T14:14:00+00:00"
      },
      "relationships": {
        "uid": {
          "data": {
            "type": "user--user",
            "id": "44418b09-7bfb-43ef-a488-7a03211bf407"
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/b657da23-243a-4c96-ae14-59971da69a6d\/relationships\/uid"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/b657da23-243a-4c96-ae14-59971da69a6d\/uid"
            }
          }
        }
      },
      "links": {
        "self": {
          "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/b657da23-243a-4c96-ae14-59971da69a6d"
        }
      }
    },
    {
      "type": "file--file",
      "id": "fc9b6db2-0c29-4337-b4a2-fed055646856",
      "attributes": {
        "drupal_internal__fid": 78,
        "langcode": "en",
        "filename": "2019-rapportage-wallengebied.pdf",
        "uri": {
          "value": "public:\/\/publications\/2019-rapportage-wallengebied.pdf",
          "url": "\/sites\/default\/files\/publications\/2019-rapportage-wallengebied.pdf"
        },
        "filemime": "application\/pdf",
        "filesize": 965788,
        "status": true,
        "created": "2019-07-10T14:13:03+00:00",
        "changed": "2019-07-10T14:14:00+00:00"
      },
      "relationships": {
        "uid": {
          "data": {
            "type": "user--user",
            "id": "44418b09-7bfb-43ef-a488-7a03211bf407"
          },
          "links": {
            "self": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/fc9b6db2-0c29-4337-b4a2-fed055646856\/relationships\/uid"
            },
            "related": {
              "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/fc9b6db2-0c29-4337-b4a2-fed055646856\/uid"
            }
          }
        }
      },
      "links": {
        "self": {
          "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/file\/file\/fc9b6db2-0c29-4337-b4a2-fed055646856"
        }
      }
    }
  ],
  "links": {
    "self": {
      "href": "http:\/\/acc.cms.data.amsterdam.nl\/jsonapi\/node\/publication?filter%5Bdrupal_internal__nid%5D=41\u0026include=field_cover_image%2Cfield_publication_file"
    }
  }
}