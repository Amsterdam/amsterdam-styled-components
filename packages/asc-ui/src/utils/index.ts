export { default as fromTheme, valueFromObject } from './fromTheme'
export {
  color,
  getColorCode,
  getTypographyFromTheme,
  focusStyleOutline,
  srOnlyStyle,
  breakpoint,
  svgFill,
  showHide,
  ShowHideTypes,
  customCss,
  CustomCssPropsType,
  CustomCssPropsInterface,
} from './themeUtils'
