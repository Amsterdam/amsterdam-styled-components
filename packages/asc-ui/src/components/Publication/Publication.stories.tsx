import React from 'react'
import { storiesOf } from '@storybook/react/'
import { action } from '@storybook/addon-actions'
import { Column, CustomHTMLBlock, Row, ArticleMetaList } from '../../index'
import Summary from '../Typography/Summary'
import ArticleHeader from '../Article/ArticleHeader/ArticleHeader'
import Downloader from '../Downloader'
import PublicationContent from './PublicationContent'
import Publication from './Publication'
import publicationJSON from './publication_feed.json'

storiesOf('Composed/Publication', module).add('with image', () => {
  const {
    title,
    // created,
    body,
    field_publication_intro: intro,
    field_file_size: fileSize,
    field_file_type: fileType,
  } = publicationJSON.data[0].attributes
  const cmsRoot = 'https://acc.cms.data.amsterdam.nl'
  const coverUrl = publicationJSON.included[0].attributes.uri.url
  const downloadUrl = publicationJSON.included[1].attributes.uri.url

  return (
    <>
      <br />
      <br />
      <Publication>
        <Row>
          <Column
            wrap
            span={{ small: 1, medium: 4, big: 6, large: 12, xLarge: 12 }}
          >
            <Column
              span={{ small: 1, medium: 4, big: 6, large: 12, xLarge: 12 }}
            >
              <PublicationContent>
                <ArticleHeader title={title}></ArticleHeader>
                <ArticleMetaList
                  fields={[
                    { id: 1, label: fileSize },
                    { id: 2, label: fileType },
                  ]}
                ></ArticleMetaList>
              </PublicationContent>
            </Column>
            <Column span={{ small: 1, medium: 4, big: 3, large: 6, xLarge: 6 }}>
              <Downloader
                imageSrc={`${cmsRoot}${coverUrl}`}
                description={`Download PDF (${fileSize})`}
                onClick={action(`downloading: ${cmsRoot}${downloadUrl}`)}
              />
            </Column>
            <Column span={{ small: 1, medium: 4, big: 3, large: 6, xLarge: 6 }}>
              <PublicationContent>
                {intro && <Summary>{intro}</Summary>}
                <CustomHTMLBlock
                  dangerouslySetInnerHTML={{
                    __html: body.processed,
                  }}
                ></CustomHTMLBlock>
              </PublicationContent>
            </Column>
          </Column>
        </Row>
      </Publication>
    </>
  )
})
