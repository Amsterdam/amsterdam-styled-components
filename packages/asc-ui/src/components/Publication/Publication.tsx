import React from 'react'
import PublicationStyle from './PublicationStyle'
import { BlogPost } from '../Blog'

type Props = {}

const Publication: React.FC<Props> = ({ children }) => (
  <BlogPost StyledComponent={PublicationStyle}>{children}</BlogPost>
)

export default Publication
