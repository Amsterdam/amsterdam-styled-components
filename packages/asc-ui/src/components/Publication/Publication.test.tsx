import * as React from 'react'
import Publication from './Publication'
import { renderWithTheme } from '../../utils/withTheme'
import 'jest-styled-components'

jest.useFakeTimers()

describe('Publication', () => {
  let component: any

  it('should render', () => {
    component = renderWithTheme(
      <Publication>This is the publication content</Publication>,
    )
    expect(component).toMatchSnapshot()
  })
})
