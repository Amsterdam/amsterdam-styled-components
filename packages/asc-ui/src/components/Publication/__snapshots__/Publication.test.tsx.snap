// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication should render 1`] = `
.c0 {
  width: 100%;
  color: #323232;
  background-color: #ffffff;
  margin-top: 20px;
  margin-bottom: 20px;
}

.c0 .c1 {
  max-height: 560px;
  min-height: 250px;
}

@media screen and (min-width:768.02px) {
  .c0 {
    margin-top: 54px;
    margin-bottom: 70px;
  }
}

<section
  class="c0"
>
  This is the publication content
</section>
`;
